test/StringToolsTests.h

```diff
@@ -0,0 +1,333 @@
+        String DupWhiteResultString("This is a needlessly wide sentence. ");
+        StringTools::RemoveDuplicateWhitespaces(DupWhiteSourceString);
+        TEST_EQUAL("RemoveDuplicateWhitespaces(String&)",DupWhiteResultString,DupWhiteSourceString);
+
+        String DupWhiteCopySourceString("\t\tThis indented  sentence   is    stretching     really      far.");
+        String DupWhiteCopyResultString("\tThis indented sentence is stretching really far.");
+        TEST_EQUAL("RemoveDuplicateWhitespacesCopy(String)",
+                   DupWhiteCopyResultString,StringTools::RemoveDuplicateWhitespacesCopy(DupWhiteCopySourceString));
+    }//Size Modifying String Manipulation
+
+    {//Colour Channels
+        StringView HexStr1("71");
+        StringView HexStr2("A2");
+        StringView HexStr3("3F");
+        StringView HexStr4("06");
+        TEST_EQUAL_EPSILON("ConvertHexToColourChannel(const_StringView)-First",
+                           Real(113.0 / 255.0),StringTools::ConvertHexToColourChannel(HexStr1));
+        TEST_EQUAL_EPSILON("ConvertHexToColourChannel(const_StringView)-Second",
+                           Real(162.0 / 255.0),StringTools::ConvertHexToColourChannel(HexStr2));
+        TEST_EQUAL_EPSILON("ConvertHexToColourChannel(const_StringView)-Third",
+                           Real(63.0 / 255.0),StringTools::ConvertHexToColourChannel(HexStr3));
+        TEST_EQUAL_EPSILON("ConvertHexToColourChannel(const_StringView)-Fourth",
+                           Real(6.0 / 255.0),StringTools::ConvertHexToColourChannel(HexStr4));
+
+        Real ChannelVal1 = 0.8675f;//221.2125
+        Real ChannelVal2 = 0.1234f;//31.467
+        Real ChannelVal3 = 0.9274f;//236.487
+        Real ChannelVal4 = 0.0475f;//12.1125
+        TEST_EQUAL("ConvertColourChannelToHex(const_Real)-First",
+                   String("DD"),StringTools::ConvertColourChannelToHex(ChannelVal1));
+        TEST_EQUAL("ConvertColourChannelToHex(const_Real)-Second",
+                   String("1F"),StringTools::ConvertColourChannelToHex(ChannelVal2));
+        TEST_EQUAL("ConvertColourChannelToHex(const_Real)-Third",
+                   String("EC"),StringTools::ConvertColourChannelToHex(ChannelVal3));
+        TEST_EQUAL("ConvertColourChannelToHex(const_Real)-Fourth",
+                   String("0C"),StringTools::ConvertColourChannelToHex(ChannelVal4));
+    }//Colour Channels
+
+    {//Tokenizing
+        String SplitSourceString("This is a string that will be split into 11 parts.");
+        StringVector DefaultSplitResult = StringTools::Split(SplitSourceString);
+        StringVector ISplitResult = StringTools::Split(SplitSourceString,"i",5);// Should normally split 7 times.
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-DefaultCount",
+                   size_t(11),DefaultSplitResult.size());
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-DefaultElement1",
```

Comment:
This mixes two different tests. It is a minor nit, but 2 sets of tests, one testing the split char and another testing split limit would better follow best practices. That said, this is simple enough its not a real issue.

Reply:
There are two sets of tests for Split as is.  One uses the default value for splits and the other doesn't.  Although the one that doesn't (the ISplit tests) also happens to use a different delimiter.  I guess it's still not ideal, but we do have one default value test and one non-default value test.